### PR TITLE
add manufacturing-specific shirt report

### PR DESCRIPTION
**Pull request by binary1230**
POLICY IMPLICATIONS NEED SANITY CHECKING ON THIS! CC @kitsuta @EliAndrewC  @ShevaDas 

for @bds002.  

There are 2 types of shirts now: (this is a change from previous years): 
       - "staff shirts" - staff uniforms, each staff (someone who has an actual staff badge) gets TWO
       - "swag shirts" - pre-ordered shirts, which the following groups receive:
           - volunteers (non-staff who get a swag shirt for free)  (NOT someone with a staff badge)
           - attendees (who get whatever they buy)

the new shirt manufacturing report shows:
1) count and sizes of staff shirts
2) count and sizes of swag pre-order shirts

this report will be given to the merch contractor (probably after we add some padding) so they know how many shirts to order

Whoever is doing this will need to pad out the number of shirts so there are safety margins.

### CODE REVIEW

**Comment by binary1230** [comment]
Brent and I are happy with the policy implications of this (NOTABLE: staffers do NOT get a free pre-order shirt anymore, they get 2 staff shirts instead).  Just want to get signoff on that.

**Comment by binary1230** [comment]
ALSO: note that there are no Shift/Job requirements for staffers, so people signed up for 0 hours are included in this count.

File: uber/site_sections/summary.py [hunk 1]
```diff
@@ -224,6 +224,42 @@ def volunteers_with_worked_hours(self, out, session):
...
+
+            # TODO: eventually extract these conditions into properties on Attendee
+            # Attendee.gets_free_shirt needs to be re-worked out.
+            gets_staff_shirt = attendee.badge_type == c.STAFF_BADGE
+            gets_swag_shirt = attendee.badge_type != c.STAFF_BADGE and (attendee.gets_paid_shirt or attendee.ribbon == c.VOLUNTEER_RIBBON)
+
+            if gets_staff_shirt:
+                counts['staff'][label(shirt_label)] += c.SHIRTS_PER_STAFFER
+
+            if attendee.gets_paid_shirt:
```
Comment (kitsuta) [inline uber/site_sections/summary.py]: Didn't you just calculate `gets_swag_shirt` a few lines above? Why not use it here?

File: uber/site_sections/summary.py [hunk 2]
```diff
@@ -224,6 +224,42 @@ def volunteers_with_worked_hours(self, out, session):
...
+        sort = lambda d: sorted(d.items(), key=lambda tup: labels.index(tup[0]))
+        label = lambda s: 'size unknown' if s == c.SHIRTS[c.NO_SHIRT] else s
+
+        for attendee in session.staffers(only_staffing=False):
+            shirt_label = attendee.shirt_label or 'size unknown'
+
+            # TODO: eventually extract these conditions into properties on Attendee
+            # Attendee.gets_free_shirt needs to be re-worked out.
+            gets_staff_shirt = attendee.badge_type == c.STAFF_BADGE
+            gets_swag_shirt = attendee.badge_type != c.STAFF_BADGE and (attendee.gets_paid_shirt or attendee.ribbon == c.VOLUNTEER_RIBBON)
```
Comment (kitsuta) [inline uber/site_sections/summary.py]: Wait, this change implies that staff do NOT get this year's themed shirt. I assume that's intended but I want to point it out just to double-check.

Comment (kitsuta) [inline uber/site_sections/summary.py]: Oops, you totally commented to that effect.

Comment (bds002) [inline uber/site_sections/summary.py]: That is correct.  
Those with Staff badges will get 2 MAGFest 2017 Staff themed shirts. 
They will not be getting an Attendee themed shirt for free.

**Comment by kitsuta** [comment]
So this is obviously way too late to comment on the policy change, but requiring that all staffers wear a staff shirt while 'on duty' is, _generally speaking_, mildly crappy to any staffer with an unusual body type. And in the t-shirt industry, 'curvy' (of the female variety) often qualifies.

I know MAGFest has tried very hard to get good, well-fitting female t-shirts in recent years but the fact is that ... well, doing that successfully is hard. I have heard personally that we have in the past tried to avoid the 'babydoll' style only to get exactly that from our vendor. I wear 0 of my MAGFest t-shirts in my day-to-day life. The t-shirt I used to wear, from M12, is now on the bottom of the t-shirt pile because I have so many other comfier (and better-fitting) shirts.

This isn't about needing to get a specific style of women's shirt (I know several women prefer the 'babydoll' style), it's about the fact that having a dress code that involves someone, somewhere, ordering the exact shirt I have to wear while on shift is just... going to be kind of a pain in the butt for a lot of people.

I won't personally be affected by this, and I think a lot of our upper tier staff also won't personally be affected by this, but I worry that our 'frontline' staff will be the most inconvenienced about it and will feel the least like they can complain about it. So since we're pretty much committed to the policy change for this year, I would definitely strongly encourage us to arrange a specific survey for all staff about how they feel about it. We can send it out right after the fest to get the freshest opinions on their actual experiences with using the staff shirts while on shift.

**Comment by bds002** [comment]
Understood, and I am happy to talk to anyone about the why's and the who's and what's about bringing back staff shirts. Code review isn't the place for that... uggh words mean things (policy implications).  I will be the first to say there is an expectation instead of a requirement (as was stated at MAGCon).

**Comment by kitsuta** [comment]
Ahhh I see, I was a little misled on the mandatory-ness of the shirts when they were called uniforms. We've talked about it more in Slack and I'm more comfortable.

Survey is still a good idea though. :P

File: uber/site_sections/summary.py (hunk 1 above)
Comment (binary1230) [inline uber/site_sections/summary.py]: hm yea you're right... i don't know why I did that.

**Pull request merged**